arbnode/blockmetadata.go

```diff
@@ -114,6 +151,11 @@ func (b *BlockMetadataFetcher) persistBlockMetadata(ctx context.Context, query [
 }
 
 func (b *BlockMetadataFetcher) Update(ctx context.Context) time.Duration {
+	if err := b.checkMetadataBackendChainId(ctx); err != nil {
+		log.Error("Error running the BlockMetadataFetcher, trying again in 10 minutes", "err", err)
+		return time.Minute * 10
+	}
```

Comment:
I prefer we only update blockmetadata after successfully verifying chain ID, in current impl if theres error in fetching chain id then nil is returned.
```suggestion
	if err := b.checkMetadataBackendChainId(ctx); err != nil {
		log.Error("Error running the BlockMetadataFetcher, trying again in 10 minutes", "err", err)
		return time.Minute * 10
	} else if !b.chainIdChecked{
		log.Error("Error verifying chainID in BlockMetadataFetcher, trying again in 10 minutes")
		return time.Minute * 10	         
	}
```

Reply:
Thanks, I've addressed this comment another way.